marklogic-data-hub/src/main/java/com/marklogic/hub/hubcentral/migration/HubCentralMigrator.java

```diff
@@ -0,0 +1,166 @@
+package com.marklogic.hub.hubcentral.migration;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.ObjectWriter;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.marklogic.client.ext.helper.LoggingObject;
+import com.marklogic.hub.HubConfig;
+import com.marklogic.hub.HubProject;
+import com.marklogic.hub.impl.EntityManagerImpl;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Stream;
+
+public class HubCentralMigrator extends LoggingObject {
+    private static final List<String> removableIndexArrays = Arrays.asList("elementRangeIndex", "rangeIndex", "pathRangeIndex");
+    private HubConfig hubConfig;
+    private EntityManagerImpl entityManager;
+    private FlowMigrator flowMigrator;
+    private ObjectMapper mapper = new ObjectMapper();
+
+    public HubCentralMigrator(HubConfig hubConfig) {
+        this.hubConfig = hubConfig;
+        this.entityManager = new EntityManagerImpl(hubConfig);
+        this.flowMigrator = new FlowMigrator(this.hubConfig);
+    }
+
+    /**
+     * Migrate the entity model, flow and mapping files in a user's local project. Does not make any changes to what's stored in MarkLogic.
+     */
+    public void migrateUserArtifacts() {
+        flowMigrator.migrateFlows();
+        migrateEntityModels();
+    }
+
+    /**
+     * Migrate the entity model files in a user's local project. Does not make any changes to what's stored in MarkLogic.
+     */
+    protected void migrateEntityModels() {
+        HubProject hubProject = hubConfig.getHubProject();
+        final File entityModelsDir = hubProject.getHubEntitiesDir().toFile();
+        if (!entityModelsDir.exists()) {
+            logger.warn("No entities directory exists, so no entity models will be migrated");
+            return;
+        }
+
+        logger.warn("Beginning migration of entity models in entities directory");
+
+        ObjectWriter writer = mapper.writerWithDefaultPrettyPrinter();
+        boolean atLeastOneEntityModelWasMigrated = false;
+
+        for (JsonNode entityModel : entityManager.getAllEntities()) {
```

Comment:
Another problem I thought of is that a user may very well have an entity model file with invalid JSON in it. Have seen it happen in support tickets. So we need for "getAllEntities" to account for a file being bad. I think modifying that to try/catch an error on each file - instead of throwing it - is a good idea (it'll be beneficial to the current clients of this method too).